File: src/main/java/org/goafabric/core/data/repository/entity/ContactPointEo.java
package org.goafabric.core.data.repository.entity;

import jakarta.persistence.*;
import org.goafabric.core.data.repository.extensions.AuditListener;

@Entity
@Table(name="contact_point")
@EntityListeners(AuditListener.class)
public class ContactPointEo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    public String id;

    public String use;
    public String system;

    @Column(name = "c_value")
    public String value;
}

File: src/main/kotlin/org/goafabric/core/organization/persistence/entity/ContactPointEo.kt
package org.goafabric.core.organization.persistence.entity

import jakarta.persistence.*
import org.goafabric.core.persistence.extensions.AuditTrailListener
import org.goafabric.core.persistence.extensions.KafkaPublisher

@Entity
@Table(name = "contact_point")
@EntityListeners(AuditTrailListener::class, KafkaPublisher::class)
class ContactPointEo(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null,

    var use: String? = null,
    var system: String? = null,

    @Column(name = "c_value")
    var value: String? = null
)
